Source/JavaScriptCore/runtime/JSStringJoiner.cpp

```diff
@@ -43,6 +43,54 @@ static inline void appendStringToData(CharacterType*& data, StringView string)
     data += string.length();
 }
 
+template<typename CharacterType>
+static inline void appendStringToDataWithOneCharacterSeparatorRepeatedly(CharacterType*& data, UChar separatorCharacter, StringView string, unsigned count)
```

Comment:
Just a small suggestion, but seems like switch on `separator.length() + string.length()` (instead of passing just the single character case) would be a bit more generic and wouldn't add any more complexity

Reply:
Yeah, right. But for now, I would like to defer this extension for now since one-character case is too dominant in the use cases. So we would like to tune one-character case specifically and significantly, and then we would like to expand our optimization for the other cases too.